sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java

```diff
@@ -562,7 +565,7 @@ protected Object decode(ChannelHandlerContext ctx, Channel channel, ChannelBuffe
                     int protocol = buffer.readInt();
                     traceLogProtocol(protocol);
                     state = State.STARTUP_BODY;
-                    return nullOrBuffer(buffer);
+                    return nullOrBuffer(buffer, previousState);
```

Comment:
instead of passing in the previousState, wouldn't it be clearer to pass in a nextState and only set it if a buffer is returned instead of null?